.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: ci
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  go:
+    strategy:
+      fail-fast: false
+      matrix:
+        container: ["cimg/go:1.14", "cimg/go:1.15"]
```

Comment:
Are we planning to stay on the circle ci images, or are we planning to switch to (for example) the setup-go action?  If the latter, we should open an issue to track that switch.

Reply:
We could do either, do you prefer to switch to the setup-go action?

Reply:
The setup-go action is usually used with github actions.  I'm not sure if the go-based container image is faster/slower than using the setup-go, which does some caching.  It looks like it also has a way to match go problems: https://github.com/actions/setup-go/blob/main/matchers.json, but i'd have to play around with it to see if it is useful.

Reply:
The setup-go action looks pretty ideal as an end state. Did we want to tackle that as a follow on PR to this?

Reply:
I can update this PR to use `setup-go` instead. should be a pretty small change. will push update shortly!

Reply:
@MrAlias updated to use setup-go!